src/states/node.rs

```diff
@@ -500,53 +524,56 @@ impl Node {
             return Err(RoutingError::UnknownConnection(peer_id));
         };
 
-        self.handle_signed_message(hop_msg.content(),
-                                   hop_msg.route(),
-                                   &hop_name,
-                                   hop_msg.sent_to())
+        let HopMessage { mut content, route, sent_to, .. } = hop_msg;
+        let hop_pub_ids = if let Some(group) = self.peer_mgr.routing_table().get_group(&hop_name) {
+            self.peer_mgr.get_pub_ids(group).into_iter().collect::<BTreeSet<_>>()
+        } else {
+            return Err(RoutingError::UnknownConnection(peer_id));
+        };
+        // TODO - If the hop group is our own group, we should add our own ID in some cases here.
+        //        However, we can't always do this (e.g. the first node of a new network won't
+        //        handle any of the second node's requests since adding its own ID will stop it
+        //        from ever accumulating).
```

Comment:
But the second node should only send the request as a `ManagedNode`, and in that case we need to consider the message accumulated as soon as it has just one signature.